pkg/controller/service/servicecontroller.go

```diff
@@ -422,10 +422,15 @@ func (s *serviceCache) delete(serviceName string) {
 }
 
 func (s *ServiceController) needsUpdate(oldService *api.Service, newService *api.Service) bool {
-	if !wantsLoadBalancer(oldService) && !wantsLoadBalancer(newService) {
+	oldServiceNeedsLoadBalancer := wantsLoadBalancer(oldService)
```

Comment:
Make sure this doesn't have adverse effects. Create a cluster with a service, leave it idle for a few days, make sure exactly 0 cloud provider mutation calls are made. 

Even with that you can only test gce. Can you confirm that it is OK for other CPs? (in theory it should be, but I don't know for sure)